app/models/company.rb

```diff
@@ -18,299 +20,21 @@ class Company
-        hiring_url: nil,
-        email:      'rspapps@ricoh.sg'
-      },
-      {
-        name:       'honestbee',
-        website:    'https://www.honestbee.com',
-        logo_url:   'https://cloud.githubusercontent.com/assets/12945452/10658532/765e7e3c-78ca-11e5-8090-892ec9912f19.png',
-        address:    '2 Alexandra Road, #04-01B, Singapore 159919',
-        hiring_url: 'http://honestbee.workable.com/',
-        email:      'edmund.yee@honestbee.com'
-      },
-      {
-        name:       'Kaligo',
-        website:    'https://www.kaligo.com',
-        logo_url:   'https://d2xr67b5foriew.cloudfront.net/assets/images/scb/kaligo-logo-red-black.png',
-        address:    '36B Mosque Street, Singapore 059514',
-        hiring_url: 'http://sg.startupjobs.asia/sg/company/1479-kaligo-singapore',
-        email:      'greg@kaligo.com'
-      },
-      {
-        name:       'SMART Centre',
-        website:    'http://smart.mit.edu',
-        logo_url:   'https://cloud.githubusercontent.com/assets/15645205/10930156/95eef638-82f6-11e5-9a55-3a1199222cf6.png',
-        address:    '1 CREATE Way, #10-01 CREATE Tower, Singapore 138602',
-        hiring_url: 'http://smart.mit.edu/jobs-at-smart/current-open-positions/article/299-software-engineer-smart-phone-based-travel-survey-.html',
-        email:      'fang.zhao@smart.mit.edu'
-      },
-      {
-        name:       'Canopy',
-        website:    'https://canopy.sg',
-        logo_url:   'https://goo.gl/ipIzwe',
-        address:    '7 Kim Yam Road, Singapore 239323',
-        hiring_url: 'mailto:admin@mesitis.com?subject=I%20can%20help%20you%20write%20better%20Ruby!',
-        email:      'admin@mesitis.com'
-      },
-      {
-        name:       'SourceClear',
-        website:    'https://srcclr.com',
-        logo_url:   'https://srcclr.com/images/srcclr__logo--primary-grey.png',
-        address:    '20 Ayer Rajah Crescent, #08-07, Singapore 139964',
-        hiring_url: 'https://jobs.lever.co/srcclr/',
-        email:      'hello@srcclr.com'
-      },
-      {
-        name:       'Grain',
-        website:    'https://grain.com.sg',
-        logo_url:   'https://grain.com.sg/grain.png',
-        address:    '82 Lorong 23 Geylang, #04-08, Singapore 388409',
-        hiring_url: 'https://grain.com.sg/jobs',
-        email:      'hello@grain.com.sg'
-      },
-      {
-        name:       'Mobile Market Monitor',
-        website:    'https://www.mobilemarketmonitor.com/',
-        logo_url:   'https://www.mobilemarketmonitor.com/assets/images/mmm-logo.png',
-        address:    '75 High Street, Singapore 179435',
-        hiring_url: 'mailto:jorge@mobilemarketmonitor.com',
-        email:      'support@mobilemarketmonitor.com'
-      }
-    ]
+      return @companies if @companies
+
+      File.open(File.join(File.dirname(__FILE__), "companies.yml")) do |file|
+        @companies = YAML.load(file).to_ary
+      end
+
+      @companies
+    end
```

Comment:
Yes, of course. I knew something wasn't quite as elegant as it should be. Will make the fix.